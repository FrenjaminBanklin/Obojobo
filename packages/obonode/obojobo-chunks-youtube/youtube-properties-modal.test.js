import React from 'react'
import { mount } from 'enzyme'

import YoutubeProperties from './youtube-properties-modal'

describe('YoutubeProperties modal', () => {
	beforeEach(() => {
		jest.clearAllMocks()
	})

	test('YoutubeProperties component renders correctly', () => {
		const component = mount(<YoutubeProperties content={{}} />)
		const tree = component.html()

		expect(tree).toMatchSnapshot()
	})

	test('YouTubeProperties component focuses on first element', () => {
		const component = mount(<YoutubeProperties content={{}} />)

		component.instance().focusOnFirstElement()
		expect(component.html()).toMatchSnapshot()
	})

	test('YoutubeProperties component changes id', () => {
		const onConfirm = jest.fn()
		const component = mount(<YoutubeProperties onConfirm={onConfirm} content={{}} />)
		// find the id input
		component
			.find('input')
			.at(1)
			.simulate('change', {
				target: {
					value: 'mockVideoId'
				}
			})

		// click confirm
		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).toHaveBeenCalled()
		expect(component.html()).toMatchSnapshot()
	})

	test('YoutubeProperties component changes startTime', () => {
		const onConfirm = jest.fn()
		const component = mount(<YoutubeProperties onConfirm={onConfirm} content={{}} />)
		// find the startTime input
		component
			.find('input')
			.at(2)
			.simulate('change', {
				target: {
					value: 100
				}
			})

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(component.html()).toMatchSnapshot()
		expect(onConfirm).toHaveBeenCalled()
	})

	test('Youtube properties changes endTime', () => {
		const onConfirm = jest.fn()
		const component = mount(<YoutubeProperties onConfirm={onConfirm} content={{}} />)

		// find the endTime input
		component
			.find('input')
			.at(3)
			.simulate('change', {
				target: {
					value: 100
				}
			})

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(component.html()).toMatchSnapshot()
		expect(onConfirm).toHaveBeenCalled()
	})

	test('YoutubeProperties component checks startTime before confirming', () => {
		const onConfirm = jest.fn()
		const component = mount(<YoutubeProperties onConfirm={onConfirm} content={{}} />)
		const spy = jest.spyOn(component.instance(), 'setState')

		component
			.find('input')
			.at(2)
			.simulate('change', {
				target: {
					value: -1000000
				}
			})

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
		expect(spy).toHaveBeenCalledWith({ startTimeError: 'Start time must be > 0' })
	})

	test('YouTubeProperties component checks endTime before confirming', () => {
		const onConfirm = jest.fn()
		const mockContent = {
			content: { startTime: 100 }
		}
		const component = mount(<YoutubeProperties onConfirm={onConfirm} content={mockContent} />)
		const spy = jest.spyOn(component.instance(), 'setState')

		component
			.find('input')
			.at(3)
			.simulate('change', {
				target: {
					value: -1000000
				}
			})

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
		expect(spy).toHaveBeenCalledWith({ endTimeError: 'End time must be > 0' })
	})

	test('YouTubeProperties component checks endTime is less than startTime', () => {
		const onConfirm = jest.fn()
		const component = mount(<YoutubeProperties onConfirm={onConfirm} content={{}} />)
		const spy = jest.spyOn(component.instance(), 'setState')

		component
			.find('input')
			.at(2)
			.simulate('change', {
				target: {
					value: 100
				}
			})

		component
			.find('input')
			.at(3)
			.simulate('change', {
				target: {
					value: 90
				}
			})

		component
			.find('button')
			.at(1)
			.simulate('click')

		expect(onConfirm).not.toHaveBeenCalled()
		expect(spy).toHaveBeenCalledWith({ endTimeError: 'End time must be > start time' })
	})
})
