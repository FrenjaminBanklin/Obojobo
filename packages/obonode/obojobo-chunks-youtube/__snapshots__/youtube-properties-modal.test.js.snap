// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouTubeProperties modal YouTubeProperties component changes endTime 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">YouTube Video</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"youtube-video-properties\\"><div class=\\"youtube-video-properties-input-wrapper\\"><label>Youtube video url:</label><div><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain youtube video options\\">?</button></div></div></div><input id=\\"obojobo-draft--chunks--youtube--video-url\\" type=\\"text\\" value=\\"\\"><input id=\\"obojobo-draft--chunks--youtube--video-id\\" type=\\"hidden\\" value=\\"\\"><label>Start time (optional):</label><input type=\\"number\\" min=\\"0\\" value=\\"100\\"><small>Seconds or MM:SS format (e.g. 135 or 2:15)</small><span class=\\"error\\"></span><label>End time (optional):</label><input type=\\"number\\" min=\\"1\\" value=\\"\\"><span class=\\"error\\"></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`YouTubeProperties modal YouTubeProperties component changes id 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">YouTube Video</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"youtube-video-properties\\"><div class=\\"youtube-video-properties-input-wrapper\\"><label>Youtube video url:</label><div><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain youtube video options\\">?</button></div></div></div><input id=\\"obojobo-draft--chunks--youtube--video-url\\" type=\\"text\\" value=\\"mockVideoId\\" class=\\"youtube--video-invalid\\"><input id=\\"obojobo-draft--chunks--youtube--video-id\\" type=\\"hidden\\" value=\\"\\"><label>Start time (optional):</label><input type=\\"number\\" min=\\"0\\" value=\\"\\"><small>Seconds or MM:SS format (e.g. 135 or 2:15)</small><span class=\\"error\\"></span><label>End time (optional):</label><input type=\\"number\\" min=\\"1\\" value=\\"\\"><span class=\\"error\\"></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`YouTubeProperties modal YouTubeProperties component changes startTime 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">YouTube Video</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"youtube-video-properties\\"><div class=\\"youtube-video-properties-input-wrapper\\"><label>Youtube video url:</label><div><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain youtube video options\\">?</button></div></div></div><input id=\\"obojobo-draft--chunks--youtube--video-url\\" type=\\"text\\" value=\\"\\"><input id=\\"obojobo-draft--chunks--youtube--video-id\\" type=\\"hidden\\" value=\\"\\"><label>Start time (optional):</label><input type=\\"number\\" min=\\"0\\" value=\\"\\"><small>Seconds or MM:SS format (e.g. 135 or 2:15)</small><span class=\\"error\\"></span><label>End time (optional):</label><input type=\\"number\\" min=\\"1\\" value=\\"\\"><span class=\\"error\\"></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`YouTubeProperties modal YouTubeProperties component focuses on first element 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">YouTube Video</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"youtube-video-properties\\"><div class=\\"youtube-video-properties-input-wrapper\\"><label>Youtube video url:</label><div><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain youtube video options\\">?</button></div></div></div><input id=\\"obojobo-draft--chunks--youtube--video-url\\" type=\\"text\\" value=\\"\\"><input id=\\"obojobo-draft--chunks--youtube--video-id\\" type=\\"hidden\\" value=\\"\\"><label>Start time (optional):</label><input type=\\"number\\" min=\\"0\\" value=\\"\\"><small>Seconds or MM:SS format (e.g. 135 or 2:15)</small><span class=\\"error\\"></span><label>End time (optional):</label><input type=\\"number\\" min=\\"1\\" value=\\"\\"><span class=\\"error\\"></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;

exports[`YouTubeProperties modal YouTubeProperties component renders correctly 1`] = `"<div class=\\"obojobo-draft--components--modal--simple-dialog\\"><div class=\\"obojobo-draft--components--modal--dialog \\"><div class=\\"obojobo-draft--components--modal--modal\\" role=\\"dialog\\" aria-labelledby=\\"obojobo-draft--components--modal--modal--content\\"><input class=\\"first-tab\\" type=\\"text\\"><div class=\\"content\\" id=\\"obojobo-draft--components--modal--modal--content\\"><h1 class=\\"heading\\" style=\\"text-align: center;\\">YouTube Video</h1><div class=\\"dialog-content\\" style=\\"text-align: center;\\"><div class=\\"youtube-video-properties\\"><div class=\\"youtube-video-properties-input-wrapper\\"><label>Youtube video url:</label><div><div class=\\"obojobo-draft--components--more-info-button is-default-label is-mode-hidden\\"><button aria-label=\\"Click to explain youtube video options\\">?</button></div></div></div><input id=\\"obojobo-draft--chunks--youtube--video-url\\" type=\\"text\\" value=\\"\\"><input id=\\"obojobo-draft--chunks--youtube--video-id\\" type=\\"hidden\\" value=\\"\\"><label>Start time (optional):</label><input type=\\"number\\" min=\\"0\\" value=\\"\\"><small>Seconds or MM:SS format (e.g. 135 or 2:15)</small><span class=\\"error\\"></span><label>End time (optional):</label><input type=\\"number\\" min=\\"1\\" value=\\"\\"><span class=\\"error\\"></span></div></div><div class=\\"controls\\"><div class=\\"obojobo-draft--components--button alt-action is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\">Cancel</button></div><div class=\\"obojobo-draft--components--button is-not-dangerous align-center\\"><button class=\\"button\\" contenteditable=\\"false\\" tabindex=\\"0\\">OK</button></div></div></div><input class=\\"last-tab\\" type=\\"text\\"></div></div></div>"`;
